### PR TITLE
Add users search api doc fix #16

### LINKED ISSUE #16: [Help Wanted] Document '/users/search' API for Mattermost 3.5
If you're interested please comment here and come [join our "Contributors" community channel](https://pre-release.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team.

For technical advice or questions, please  [join our "Developers" community channel](https://pre-release.mattermost.com/core/channels/developers).

---

**Details:**
The API endpoint `/api/v3/users/search` is undocumented and we would love help getting it documented.

Please look at the [release-3.5 branch of mattermost/platform](https://github.com/mattermost/platform/tree/release-3.5/api) when documenting this API. Looking at our [Go driver](https://github.com/mattermost/platform/blob/release-3.5/model/client.go) and [JavaScript driver](https://github.com/mattermost/platform/blob/release-3.5/webapp/client/client.jsx) might help as well.

Also please see [CONTRIBUTING.md](https://github.com/mattermost/mattermost-api-reference/blob/master/CONTRIBUTING.md) for help with contributing to this repo. If you have any questions, feel free to ask!

### PR DESCRIPTION
**Pull request by MartinDelille**
Not so sure about the description

### CODE REVIEW

**Comment by MartinDelille** [comment]
ref #16

File: source/users.yaml [hunk 1]
```diff
@@ -154,6 +154,41 @@
...
+      parameters:
+        - in: body
+          name: body
+          description: Search parameter
+          required: true
+          schema:
+            type: object
+            required:
+              - term
+              - team_id
```
Comment (jwilander) [inline source/users.yaml]: `team_id` is actually optional, when it's blank or missing the search is performed across all teams

File: source/users.yaml [hunk 2]
```diff
@@ -154,6 +154,41 @@
...
+                type: string
+              in_channel_id:
+                 type: string
+              not_in_channel_id:
+                type: string
+              allow_inactive:
+                type: boolean
+      responses:
+        '200':
+          description: Profiles retreived successfully
```
Comment (jwilander) [inline source/users.yaml]: Minor typo, should be `retrieved`

File: source/users.yaml [hunk 3]
```diff
@@ -154,6 +154,41 @@
             additionalProperties:
               $ref: "#/definitions/User"
 
+  '/users/search':
+    post:
+      tags:
+        - users
+      summary: Search users
+      description: Get a map of sanitized user objects by search parameter
```
Comment (jwilander) [inline source/users.yaml]: Maybe `Get a list of user objects matching some search parameters.`?

**Comment by MartinDelille** [comment]
Done

**Comment by jwilander** [comment]
Awesome, thanks!

**Pull request merged**